contracts/Comptroller.sol

```diff
@@ -1238,6 +1242,13 @@ contract Comptroller is ComptrollerV6Storage, ComptrollerInterface, ComptrollerE
      * @param borrower The address of the borrower to distribute COMP to
      */
     function distributeBorrowerComp(address cToken, address borrower, Exp memory marketBorrowIndex) internal {
+        // Don't distribute borrower COMP if the user is not in the borrower market.
+        // The user's borrow state doesn't have to be updated if they're not borrowing.
+        // A user's borrow state is updated before borrowing and also after repaying; this sufficiently keeps the
+        // user's borrow state up-to-date.
+        if (!markets[cToken].accountMembership[borrower])
+            return;
```

Comment:
Reduces gas cost of claim by 35K. A net change from the prior commit by 8K. :partying_face: